### PR TITLE
Add a double/float variation of sarray_sort

### PR DESCRIPTION
**Pull request by thilinarmtb**
This PR sorts a double/float array locally.

### TODO
- [x] Add tests
- [ ] Figure out how to select the type of input (`long, int, double, float`)

### CODE REVIEW

File: src/sarray_sort.h [hunk 1]
```diff
@@ -46,10 +46,14 @@ void sarray_permute_buf_(
...
-  if(is_long) \
+  if(is_long == 1) \
     sortp_long(buf,keep, (ulong*)((char*)(A)+offsetof(T,field)),n,sizeof(T)); \
-  else \
+  else if(is_long == 0) \
     sortp     (buf,keep, (uint *)((char*)(A)+offsetof(T,field)),n,sizeof(T)); \
+  else if(is_long == 3) \
+    sortp_double(buf,keep, (double *)((char*)(A)+offsetof(T,field)),n,sizeof(T)); \
+  else if(is_long == 2) \
+    sortp_float(buf,keep, (float *)((char*)(A)+offsetof(T,field)),n,sizeof(T)); \
```
Comment (thilinarmtb) [inline src/sarray_sort.h]: The variable `is_long` should be renamed to a something like `type`.
Currently,
```sh
is_long =1 --> unsigned long
is_long = 0 --> unsigned int
is_long = 3 --> double
is_long = 4 --> float
```

**Comment by thilinarmtb** [comment]
@stgeke : I think this is good to merge as it is. My only concern is switching between types.
Previously, the parameter `is_long` was used to pick an implementation from `long` or an `int`.`
All the non-zero values of is_long was used to pick the long sort.

In this PR I am suggesting the use of `is_long = 3` for `double` and `is_long = 2` for `float`. I am
not sure if this will break existing user code, i.e., instances where user used `is_long = 2` or `3` to
select the long implementation.

This functionality is needed for the new genmap implementation.

**Comment by stgeke** [comment]
Can you please test if this change breaks existing user code like Nek5000

**Comment by thilinarmtb** [comment]
Closed in favor of #30

**Pull request closed without merging**